### PR TITLE
✨ Duplicate&ToTemplate in frontend

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?

- [x] Show progress and status of copy operation
- [x] Show progress and status of template creation
- [x] Retry call to task status
- [x] Abort task (cancelling the task by calling the /abort endpoint)
- [x] Retrieve currently running tasks from GET /tasks endpoint

Duplicate

## Related issue/s
closes https://github.com/ITISFoundation/osparc-simcore/issues/3326

### CODE REVIEW

**Review by pcrespov — COMMENTED** [review]
- MINOR: I think "successfully aborted" message should have a distinctive icon and not just the tick (something like https://fonts.google.com/icons?icon.query=cancel&icon.platform=web ? )
- Q: When you logout, there are some "unexpected" errors popping up ... what are those? If there are some tasks in the background, should you warn the user before logging out whether he/she wants to keep those running or abort them?

File: services/static-webserver/client/source/class/osparc/desktop/MainPage.js [hunk 1]
```diff
@@ -190,8 +190,34 @@ qx.Class.define("osparc.desktop.MainPage", {
           }, this);
         }
       });
+      studyBrowser.addListener("publishTemplate", e => this.__publishTemplate(e.getData()));
+    },
+
+    __publishTemplate: function(data) {
+      const text = this.tr("Templating process started and added to the background tasks");
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/desktop/MainPage.js:197]: I'd change "Templating process started and added to the background tasks"" with "Started template creation and added to the background tasks"

**Review by elisabettai — APPROVED** [review]
Woooow, super nice! I want to clone the whole oSPARC now. (: 

Is the progress bar actually progressing? I wasn't able to see it advancing n the animation.

File: services/static-webserver/client/source/class/osparc/desktop/MainPage.js (hunk 1 above)
Comment (odeimaiz) [reply services/static-webserver/client/source/class/osparc/desktop/MainPage.js]: Done 👌

**Comment by odeimaiz** [comment]
> Woooow, super nice! I want to clone the whole oSPARC now. (:
> 
> Is the progress bar actually progressing? I wasn't able to see it advancing n the animation.

Re the Progress Bar, yes it does progress (:, but storage was kaputt when I recorded the animation...

**Comment by odeimaiz** [comment]
> * MINOR: I think "successfully aborted" message should have a distinctive icon and not just the tick (something like https://fonts.google.com/icons?icon.query=cancel&icon.platform=web ? )

I'll have a look.

> * Q: When you logout, there are some "unexpected" errors popping up ... what are those? If there are some tasks in the background, should you warn the user before logging out whether he/she wants to keep those running or abort them?

I also realized when I recorded it. Now, when a user logs out or when he/she gets logged out, all the polling tasks are cancelled.

**Review by sanderegg — APPROVED** [review]
it is pure awesomness!

**Pull request merged**